[Bug]: Version workflow to auto update input manifests is failing

### Describe the bug

Our version workflow to auto raise PR for input manifest update is currently failing for almost 10 days. https://github.com/opensearch-project/opensearch-build/actions/workflows/versions.yml
The context of error:
```
BUILD FAILED in 11s
2022-11-04 00:49:45 WARNING  Error getting version of performance-analyzer: Command './gradlew properties -Dopensearch.version=3.0.0 -Dbuild.snapshot=false' returned non-zero exit status 1., ignored
2022-11-04 00:49:45 INFO     Checking out opensearch-observability#main
2022-11-04 00:49:45 INFO     Executing "git init" in /tmp/tmpco2c3n05/opensearch-observability
2022-11-04 00:49:45 INFO     Executing "git remote add origin https://github.com/opensearch-project/opensearch-observability.git" in /tmp/tmpco2c3n05/opensearch-observability
2022-11-04 00:49:45 INFO     Executing "git fetch --depth 1 origin main" in /tmp/tmpco2c3n05/opensearch-observability
2022-11-04 00:49:45 INFO     Removing /tmp/tmpco2c3n05
```
The problem might be the workflow trying to clone `https://github.com/opensearch-project/opensearch-observability.git` but the correct URL should be `https://github.com/opensearch-project/observability.git`
### To reproduce

https://github.com/opensearch-project/opensearch-build/actions/workflows/versions.yml

### Expected behavior

_No response_

### Screenshots

If applicable, add screenshots to help explain your problem.

### Host / Environment

_No response_

### Additional context

_No response_

### Relevant log output

_No response_

--- a/src/manifests/component_manifest.py
+++ b/src/manifests/component_manifest.py
@@ -41,12 +41,16 @@ class ComponentManifest(Manifest[ManifestType], Generic[ManifestType, Components
 
 class Components(Dict[str, ComponentType], Generic[ComponentType]):
     def __init__(self, data: Dict[Any, Any]) -> None:
-        create_component: Callable[[Any], Tuple[str, ComponentType]] = lambda component: (component["name"], self.__create__(component))
+        create_component: Callable[[Any], Tuple[str, ComponentType]] = lambda component: (
+        component["name"], self.__create__(component))
         super().__init__(map(create_component, data))
 
     @classmethod
     def __create__(self, data: dict) -> ComponentType:
-        return Component(data)  # type: ignore[return-value]
+        if "repository" in data:
+            return ComponentFromSource(data)  # type: ignore[return-value]
+        else:
+            return Component(data)
 
     def __to_dict__(self) -> List[Dict[Any, Any]]:
         as_dict: Callable[[ComponentType], dict] = lambda component: component.__to_dict__()
@@ -98,3 +102,14 @@ class Component:
             logging.info(f"Skipping {self.name}")
 
         return matches
+
+
+class ComponentFromSource(Component):
+    repository: str
+
+    def __init__(self, data: dict) -> None:
+        super().__init__(data)
+        self.repository = data["repository"]
+
+    def __to_dict__(self) -> dict:
+        return {"name": self.name, "repository": self.repository}

--- a/src/manifests/component_manifest.py
+++ b/src/manifests/component_manifest.py
@@ -42,7 +42,7 @@ class ComponentManifest(Manifest[ManifestType], Generic[ManifestType, Components
 class Components(Dict[str, ComponentType], Generic[ComponentType]):
     def __init__(self, data: Dict[Any, Any]) -> None:
         create_component: Callable[[Any], Tuple[str, ComponentType]] = lambda component: (
-        component["name"], self.__create__(component))
+            component["name"], self.__create__(component))
         super().__init__(map(create_component, data))
 
     @classmethod

--- a/src/manifests/component_manifest.py
+++ b/src/manifests/component_manifest.py
@@ -50,7 +50,7 @@ class Components(Dict[str, ComponentType], Generic[ComponentType]):
         if "repository" in data:
             return ComponentFromSource(data)  # type: ignore[return-value]
         else:
-            return Component(data)
+            return Component(data)  # type: ignore[return-value]
 
     def __to_dict__(self) -> List[Dict[Any, Any]]:
         as_dict: Callable[[ComponentType], dict] = lambda component: component.__to_dict__()

--- a/src/manifests_workflow/component_opensearch.py
+++ b/src/manifests_workflow/component_opensearch.py
@@ -27,7 +27,7 @@ class ComponentOpenSearch(Component):
         working_directory: str = None,
     ) -> 'ComponentOpenSearch':
         with GitRepository(
-            f"https://github.com/opensearch-project/{name}.git",
+            repo_url,
             branch,
             path,
             working_directory,

--- a/src/manifests_workflow/component_opensearch.py
+++ b/src/manifests_workflow/component_opensearch.py
@@ -21,6 +21,7 @@ class ComponentOpenSearch(Component):
         name: str,
         path: str,
         opensearch_version: str,
+        repo_url: str,
         branch: str = "main",
         snapshot: bool = False,
         working_directory: str = None,

--- a/src/manifests_workflow/input_manifests.py
+++ b/src/manifests_workflow/input_manifests.py
@@ -14,6 +14,7 @@ from typing import Dict, List, Type, Union
 
 import ruamel.yaml
 
+from manifests.component_manifest import ComponentFromSource
 from manifests.input_manifest import InputComponents, InputManifest
 from manifests.manifests import Manifests
 from manifests_workflow.component_opensearch import ComponentOpenSearch
@@ -99,21 +100,23 @@ class InputManifests(Manifests):
                     if component.name == self.name:
                         continue
 
-                    logging.info(f"Checking out {component.name}#main")
-                    component = component_klass.checkout(
-                        name=component.name,
-                        path=os.path.join(work_dir.name, component.name),
-                        opensearch_version=manifest.build.version,
-                        branch="main",
-                    )
+                    if type(component) is ComponentFromSource:
+                        logging.info(f"Checking out {component.name}#main")
+                        component = component_klass.checkout(
+                            name=component.name,
+                            path=os.path.join(work_dir.name, component.name),
+                            opensearch_version=manifest.build.version,
+                            repo_url=component.repository,
+                            branch="main",
+                        )
 
-                    component_version = component.version
-                    if component_version:
-                        release_version = ".".join(component_version.split(".")[:3])
-                        if release_version not in main_versions.keys():
-                            main_versions[release_version] = []
-                        main_versions[release_version].append(component)
-                        logging.info(f"{component.name}#main is version {release_version} (from {component_version})")
+                        component_version = component.version
+                        if component_version:
+                            release_version = ".".join(component_version.split(".")[:3])
+                            if release_version not in main_versions.keys():
+                                main_versions[release_version] = []
+                            main_versions[release_version].append(component)
+                            logging.info(f"{component.name}#main is version {release_version} (from {component_version})")
 
             # summarize
             logging.info("Found versions on main:")

--- a/tests/tests_manifests_workflow/test_component_opensearch.py
+++ b/tests/tests_manifests_workflow/test_component_opensearch.py
@@ -16,7 +16,7 @@ class TestComponentOpenSearch(unittest.TestCase):
     @patch("os.makedirs")
     @patch.object(GitRepository, "__checkout__")
     def test_checkout(self, *mocks: MagicMock) -> None:
-        component = ComponentOpenSearch.checkout("common-utils", "path", "1.1.0")
+        component = ComponentOpenSearch.checkout("common-utils", "path", "1.1.0", "git")
         self.assertEqual(component.name, "common-utils")
         self.assertFalse(component.snapshot)
 

--- a/tests/tests_manifests_workflow/test_input_manifests_opensearch.py
+++ b/tests/tests_manifests_workflow/test_input_manifests_opensearch.py
@@ -9,7 +9,7 @@ import os
 import unittest
 from unittest.mock import MagicMock, call, patch
 
-from manifests.input_manifest import Component
+from manifests.component_manifest import Component, ComponentFromSource
 from manifests_workflow.input_manifests_opensearch import InputManifestsOpenSearch
 
 
@@ -49,7 +49,8 @@ class TestInputManifestsOpenSearch(unittest.TestCase):
         mock_component_opensearch.return_value = MagicMock(name="common-utils")
         mock_component_opensearch.checkout.return_value = MagicMock(version="0.10.0")
         mock_input_manifest_from_path.return_value.components = {
-            "common-utils": Component({"name": "common-utils", "repository": "git", "ref": "ref"})
+            "common-utils": ComponentFromSource({"name": "common-utils", "repository": "git", "ref": "ref"}),
+            "job-scheduler": Component({"name": "job-scheduler", "dist": "zip"})
         }
         manifests = InputManifestsOpenSearch()
         manifests.update()

